### PR TITLE
CI: Build minimal image with conan layer

### PR DESCRIPTION
**Pull request by danimtb**
Added the meta-conan layer to the build of the Yocto image to check that i actually works!

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -40,7 +40,12 @@ node('Linux') {
...
-            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd poky && git clone --branch ${branchName} https://github.com/conan-io/meta-conan.git && git clone --branch ${branchName} https://github.com/openembedded/meta-openembedded.git'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-oe"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-python"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-conan"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal build'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-oe'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-python'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-conan'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal build'"
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -40,7 +40,12 @@ node('Linux') {
...
             sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir poky/conf && touch poky/conf/sanity.conf'"
-            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd poky && git clone --branch ${branchName} https://github.com/conan-io/meta-conan.git && git clone --branch ${branchName} https://github.com/openembedded/meta-openembedded.git'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-oe'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-python'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-conan'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ${PWD}/../poky/oe-init-build-env && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-oe'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ${PWD}/../poky/oe-init-build-env && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-python'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ${PWD}/../poky/oe-init-build-env && bitbake-layers add-layer ${PWD}/../poky/meta-openembedded/meta-conan'"
```

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -40,7 +40,13 @@ node('Linux') {
             sh "docker run -d -t --name yocto-builder-${env.BUILD_ID} conan-io/yocto-builder:${env.BUILD_ID}"
             sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'git clone --branch ${branchName} --depth 1 git://git.yoctoproject.org/poky'"
             sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir poky/conf && touch poky/conf/sanity.conf'"
-            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd poky && git clone --branch ${branchName} https://github.com/conan-io/meta-conan.git && git clone --branch ${branchName} https://github.com/openembedded/meta-openembedded.git'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'git clone --branch ${branchName} https://github.com/conan-io/meta-conan.git poky/meta-conan && git clone --branch ${branchName} https://github.com/openembedded/meta-openembedded.git poky/meta-openembedded'"
```

File: .ci/Jenkinsfile [hunk 4]
```diff
@@ -40,7 +40,13 @@ node('Linux') {
             sh "docker run -d -t --name yocto-builder-${env.BUILD_ID} conan-io/yocto-builder:${env.BUILD_ID}"
             sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'git clone --branch ${branchName} --depth 1 git://git.yoctoproject.org/poky'"
             sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir poky/conf && touch poky/conf/sanity.conf'"
-            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'git clone --branch ${branchName} https://github.com/conan-io/meta-conan.git poky/meta-conan && git clone --branch ${branchName} https://github.com/openembedded/meta-openembedded.git poky/meta-openembedded'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
```
Comment (danimtb) [inline .ci/Jenkinsfile:46]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash ls -la poky'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash ls -la poky/meta-openembedded'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
```

File: .ci/Jenkinsfile [hunk 5]
```diff
@@ -40,7 +40,15 @@ node('Linux') {
...
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky/meta-openembedded'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/build/poky/meta-openembedded/meta-oe'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/build/poky/meta-openembedded/meta-python'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/build/poky/meta-openembedded/meta-conan'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-oe'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-python'"
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-conan'"
```

File: .ci/Jenkinsfile [hunk 6]
```diff
@@ -40,7 +40,15 @@ node('Linux') {
...
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky/meta-openembedded'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-oe'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-python'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-conan'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-conan'"
```

File: .ci/Jenkinsfile [hunk 7]
```diff
@@ -40,7 +40,15 @@ node('Linux') {
...
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'ls -la poky/meta-openembedded'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'mkdir build'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-oe'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-openembedded/meta-python'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'cd build && source ../poky/oe-init-build-env && bitbake-layers add-layer /home/conan/poky/meta-conan'"
+            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal build'"
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
            sh "docker exec -t yocto-builder-${env.BUILD_ID} /bin/bash -c 'source poky/oe-init-build-env && bitbake -k core-image-minimal'"
```

File: .ci/Jenkinsfile [hunk 8]
```diff
@@ -14,21 +14,21 @@ node('Linux') {
...
+    //             git clone --depth 1 git://git.yoctoproject.org/poky -b ${branchName} /tmp/poky
+    //             git clone --depth 1 git://git.openembedded.org/meta-openembedded.git -b ${branchName} /tmp/meta-openembedded
+    //             cd /tmp/poky/
+    //             . /tmp/poky/oe-init-build-env
+    //             touch conf/sanity.conf
+    //             cd /tmp/meta-openembedded
+    //             yocto-check-layer -d --dependency meta-oe meta-python --with-software-layer-signature-check ${WORKSPACE}
+    //         """
+    //     }
+    // }
```
Comment (danimtb) [inline .ci/Jenkinsfile]:
```suggestion
    stage('Validate Yocto layer') {
        image.inside {
            sh """
                whoami
                WORKSPACE=${PWD}
                git clone --depth 1 git://git.yoctoproject.org/poky -b ${branchName} /tmp/poky
                git clone --depth 1 git://git.openembedded.org/meta-openembedded.git -b ${branchName} /tmp/meta-openembedded
                cd /tmp/poky/
                . /tmp/poky/oe-init-build-env
                touch conf/sanity.conf
                cd /tmp/meta-openembedded
                yocto-check-layer -d --dependency meta-oe meta-python --with-software-layer-signature-check ${WORKSPACE}
            """
        }
    }
```

**Review by uilianries — APPROVED** [review]
Nice! Only took 5h23min to build! Maybe a future feature to cache the build folder 🤔

**Pull request merged**